crates/cli-support/src/js/mod.rs

```diff
@@ -328,174 +325,28 @@ impl<'a> Context<'a> {
-
-                const wasmUrl = new URL('{module_name}_bg.wasm', import.meta.url);
-                const wasmBytes = readFileSync(wasmUrl);
-                const wasmModule = new WebAssembly.Module(wasmBytes);
-                const wasm = new WebAssembly.Instance(wasmModule, imports).exports;
-                export {{ wasm as __wasm }};\
-                "
-            ));
-        } else {
-            shim.push_str(&format!(
-                "
-                const wasmPath = `${{__dirname}}/{module_name}_bg.wasm`;
-                const wasmBytes = require('fs').readFileSync(wasmPath);
-                const wasmModule = new WebAssembly.Module(wasmBytes);
-                const wasm = exports.__wasm = new WebAssembly.Instance(wasmModule, imports).exports;\
-                "
-            ));
-        }
-
-        reset_indentation(&shim)
-    }
-
-    // generates something like
-    // ```js
-    // import * as import0 from './snippets/.../inline1.js';
-    // ```,
-    //
-    // ```js
-    // const imports = {
-    //   __proto__: null,
-    //   "./{module_name}_bg.js": {
-    //     __wbindgen_throw: function(..) { .. },
-    //     ..
-    //   },
-    //   './snippets/deno-65e2634a84cc3c14/inline1.js': import0,
-    // }
-    // ```
-    fn generate_deno_imports(&mut self, module_name: &str) -> (String, String) {
-        let mut imports = String::new();
-        let mut wasm_import_object = "const imports = {\n  __proto__: null,\n".to_string();
-        let self_module_name = format!("./{module_name}_bg.js");
-
-        wasm_import_object.push_str(&format!("  './{module_name}_bg.js': {{\n"));
+        let body = std::mem::take(&mut self.globals);
+        let body = body.trim();
 
-        for (id, js) in iter_by_import(&self.wasm_import_definitions, self.module) {
-            let import = self.module.imports.get_mut(*id);
-            wasm_import_object.push_str(&format!("{}: {},\n", &import.name, js.trim()));
-            import.module = self_module_name.clone();
+        if self.config.typescript && !self.config.mode.no_modules() && !self.config.mode.bundler() {
+            // jsr-self-types directive
+            self.globals
+                .push_str(&format!(r#"/* @ts-self-types="./{module_name}.d.ts" */"#));
+            self.globals.push('\n');
         }
 
-        wasm_import_object.push_str("\t},\n");
-
-        // e.g. snippets without parameters
-        let import_modules = self
-            .module
-            .imports
-            .iter()
-            .map(|import| &import.module)
-            .filter(|module| module.as_str() != self_module_name);
-        for (i, module) in import_modules.enumerate() {
-            imports.push_str(&format!("import * as import{i} from '{module}'\n"));
-            wasm_import_object.push_str(&format!("  '{module}': import{i},\n"))
+        if !self.js_imports.is_empty() {
+            self.globals.push_str("//#region js imports\n");
```

Comment:
While I appreciate these region comments are useful to understand the output and for debugging it, we shouldn't leave these in for user-generated bindings.

Reply:
Done